src/nv_ingest/util/nim/yolox.py

```diff
@@ -172,14 +197,41 @@ def parse_output(self, response: Any, protocol: str, data: Optional[Dict[str, An
             return response  # For gRPC, response is already a numpy array
         elif protocol == "http":
             logger.debug("Parsing output from HTTP Yolox model")
+
+            is_legacy_version = self._is_version_early_access_legacy_api()
+
             # Convert JSON response to numpy array similar to gRPC response
-            batch_results = response.get("data", [])
+            if is_legacy_version:
+                # Convert response data to GA API format.
+                response_data = response.get("data", [])
+                batch_results = []
+                for idx, detections in enumerate(response_data):
+                    curr_batch = {"index": idx, "bounding_boxes": {}}
+                    for obj in detections:
+                        obj_type = obj.get("type", "")
+                        bboxes = obj.get("bboxes", [])
+                        if not obj_type:
+                            continue
+                        if obj_type not in curr_batch:
+                            curr_batch["bounding_boxes"][obj_type] = []
+                        curr_batch["bounding_boxes"][obj_type].extend(bboxes)
+                    batch_results.append(curr_batch)
+            else:
+                batch_results = response.get("data", [])
+
             batch_size = len(batch_results)
             processed_outputs = []
 
             scaling_factors = data.get("scaling_factors", [(1.0, 1.0)] * batch_size)
 
-            for idx, detections in enumerate(batch_results):
+            x_min_label = "xmin" if is_legacy_version else "x_min"
+            y_min_label = "ymin" if is_legacy_version else "y_min"
+            x_max_label = "xmax" if is_legacy_version else "x_max"
+            y_max_label = "ymax" if is_legacy_version else "y_max"
+            confidence_label = "confidence"
+
+            for detections in batch_results:
+                idx = int(detections["index"])
                 scale_factor_x, scale_factor_y = scaling_factors[idx]
                 image_width = 1024
```

Comment:
I know its not part of your PR here, but can you make these values global constants, like YOLOX_IMAGE_PREPROC_WIDTH / YOLOX_IMAGE_PREPROC_HEIGHT

Reply:
Updated in [8ad819a](https://github.com/NVIDIA/nv-ingest/pull/270/commits/8ad819a25bf4846f9962564d6c6f70a9b56cccad).